homeassistant/components/http.py

```diff
@@ -384,18 +389,23 @@ def handle_request(self, request, **values):
         authenticated = False
 
         if self.hass.wsgi.api_password is None:
+            _LOGGER.info('Successful login from %s', request.remote_addr)
             authenticated = True
 
         elif hmac.compare_digest(request.headers.get(HTTP_HEADER_HA_AUTH, ''),
                                  self.hass.wsgi.api_password):
             # A valid auth header has been set
+            _LOGGER.info('Successful login from %s', request.remote_addr)
             authenticated = True
 
         elif hmac.compare_digest(request.args.get(DATA_API_PASSWORD, ''),
                                  self.hass.wsgi.api_password):
+            _LOGGER.info('Successful login from %s', request.remote_addr)
```

Comment:
Instead of repeating the same log message three times with the risk of them getting out of sync, let's add it before the `if self.requires_auth` line:

```python
if authenticated:
    _LOGGER.info(…)
elif self.requires_auth:
    _LOGGER.warning(…)
    raise Unauthorized()